activesupport/lib/active_support/broadcast_logger.rb

```diff
@@ -0,0 +1,196 @@
+  #
+  #   broadcast.info("Hello world!") # Writes the log to STDOUT and the development.log file.
+  #
+  # Modifying the log level to all broadcasted loggers.
+  #
+  #   stdout_logger = Logger.new(STDOUT)
+  #   file_logger   = Logger.new("development.log")
+  #   broadcast = BroadcastLogger.new
+  #   broadcast.broadcast_to(stdout_logger, file_logger)
+  #
+  #   broadcast.level = Logger::FATAL # Modify the log level for the whole broadcast.
+  #
+  # Stop broadcasting log to a sink.
+  #
+  #   stdout_logger = Logger.new(STDOUT)
+  #   file_logger   = Logger.new("development.log")
+  #   broadcast = BroadcastLogger.new
+  #   broadcast.broadcast_to(stdout_logger, file_logger)
+  #   broadcast.info("Hello world!") # Writes the log to STDOUT and the development.log file.
+  #
+  #   broadcast.stop_broadcasting_to(file_logger)
+  #   broadcast.info("Hello world!") # Writes the log *only* to STDOUT.
+  #
+  # At least one sink has to be part of the broadcast. Otherwise, your logs will not
+  # be written anywhere. For instance:
+  #
+  #   broadcast = BroadcastLogger.new
+  #   broadcast.info("Hello world") # The log message will appear nowhere.
+  #
+  # ====== A note on tagging logs while using the Broadcast logger
+  #
+  # It is quite frequent to tag logs using the `ActiveSupport::TaggedLogging` module
+  # while also broadcasting them (the default Rails.logger in development is
+  # configured in such a way).
+  # Tagging your logs can be done for the whole broadcast or for each sink independently.
+  #
+  # Tagging logs for the whole broadcast
+  #
+  #   broadcast = BroadcastLogger.new.extend(ActiveSupport::TaggedLogging)
+  #   broadcast.broadcast_to(stdout_logger, file_logger)
+  #   broadcast.tagged("BMX") { broadcast.info("Hello world!") }
+  #
+  #   Outputs: "[BMX] Hello world!" is written on both STDOUT and in the file.
+  #
+  # Tagging logs for a single logger
+  #
+  #   stdout_logger.extend(ActiveSupport::TaggedLogging)
+  #   stdout_logger.push_tags("BMX")
+  #   broadcast = BroadcastLogger.new
+  #   broadcast.broadcast_to(stdout_logger, file_logger)
+  #   broadcast.info("Hello world!")
+  #
+  #   Outputs: "[BMX] Hello world!" is written on STDOUT
+  #   Outputs: "Hello world!"       is written in the file
+  #
+  # Adding tags for the whole broadcast and adding extra tags on a specific logger
+  #
+  #   stdout_logger.extend(ActiveSupport::TaggedLogging)
+  #   stdout_logger.push_tags("BMX")
+  #   broadcast = BroadcastLogger.new.extend(ActiveSupport::TaggedLogging)
+  #   broadcast.broadcast_to(stdout_logger, file_logger)
+  #   broadcast.tagged("APP") { broadcast.info("Hello world!") }
+  #
+  #   Outputs: "[APP][BMX] Hello world!" is written on STDOUT
+  #   Outputs: "[APP] Hello world!"      is written in the file
+  class BroadcastLogger < Logger
+    # @return [Array<Logger>] All the logger that are part of this broadcast.
+    attr_reader :broadcasts
+
+    def initialize(logdev = File::NULL, *args, **kwargs)
```

Comment:
Ideally would have preferred to be able to do `BroadcastLogger.new(logger1, logger2)`, but that would mean either:

- Modify the original signature
- Create a PORO and implement duck typing because I felt the BroadcastLogger should behave just like a logger.

Not sure what's the best option

Reply:
I think we should do the latter.

Reply:
Sounds good thanks. So I guess if there is no red flag on this implementation, I'll go ahead and finish the PR (the AS test suite is passing already, I don't think this PR will change a lot to have a green CI).

Regarding 

> Broadcasting is in a grey zone and not referenced in our docs but I saw it often in apps and libraries. This refactor would break them.

Should we go through a deprecation cycle or nop?

Reply:
nop. The `broadcast` method is nodoced.

I have no reservations about this implementation. I think it will be way easier to use and maintain.